crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -0,0 +1,881 @@
+/// Entry in a manifest list.
+#[derive(Debug, PartialEq, Clone)]
+pub struct ManifestListEntry {
+    /// field: 500
+    ///
+    /// Location of the manifest file
+    manifest_path: String,
+    /// field: 501
+    ///
+    /// Length of the manifest file in bytes
+    manifest_length: i64,
+    /// field: 502
+    ///
+    /// ID of a partition spec used to write the manifest; must be listed
+    /// in table metadata partition-specs
+    partition_spec_id: i32,
+    /// field: 517
+    ///
+    /// The type of files tracked by the manifest, either data or delete
+    /// files; 0 for all v1 manifests
+    content: ManifestContentType,
+    /// field: 515
+    ///
+    /// The sequence number when the manifest was added to the table; use 0
+    /// when reading v1 manifest lists
+    sequence_number: i64,
+    /// field: 516
+    ///
+    /// The minimum data sequence number of all live data or delete files in
+    /// the manifest; use 0 when reading v1 manifest lists
+    min_sequence_number: i64,
+    /// field: 503
+    ///
+    /// ID of the snapshot where the manifest file was added
+    added_snapshot_id: i64,
+    /// field: 504
+    ///
+    /// Number of entries in the manifest that have status ADDED, when null
+    /// this is assumed to be non-zero
+    added_data_files_count: Option<i32>,
+    /// field: 505
+    ///
+    /// Number of entries in the manifest that have status EXISTING (0),
+    /// when null this is assumed to be non-zero
+    existing_data_files_count: Option<i32>,
+    /// field: 506
+    ///
+    /// Number of entries in the manifest that have status DELETED (2),
+    /// when null this is assumed to be non-zero
+    deleted_data_files_count: Option<i32>,
+    /// field: 512
+    ///
+    /// Number of rows in all of files in the manifest that have status
+    /// ADDED, when null this is assumed to be non-zero
+    added_rows_count: Option<i64>,
+    /// field: 513
+    ///
+    /// Number of rows in all of files in the manifest that have status
+    /// EXISTING, when null this is assumed to be non-zero
+    existing_rows_count: Option<i64>,
+    /// field: 514
+    ///
+    /// Number of rows in all of files in the manifest that have status
+    /// DELETED, when null this is assumed to be non-zero
+    deleted_rows_count: Option<i64>,
+    /// field: 507
+    /// element_field: 508
+    ///
+    /// A list of field summaries for each partition field in the spec. Each
+    /// field in the list corresponds to a field in the manifest file’s
+    /// partition spec.
+    partitions: Vec<FieldSummary>,
+    /// field: 519
+    ///
+    /// Implementation-specific key metadata for encryption
+    key_metadata: Vec<u8>,
+}
+
+/// The type of files tracked by the manifest, either data or delete files; Data(0) for all v1 manifests
+#[derive(Debug, PartialEq, Clone)]
+pub enum ManifestContentType {
+    /// The manifest content is data.
+    Data = 0,
+    /// The manifest content is deletes.
+    Deletes = 1,
+}
```

Comment:
I think you have to distinguish between position deletes = 1 and equality deletes = 2. If we use the `serde_repr` crate we could directly serialize/deserialize it as follows:

```rust
/// The type of files tracked by the manifest, either data or delete files; Data(0) for all v1 manifests
#[derive(Debug, Serialize_repr, Deserialize_repr, PartialEq, Eq, Clone)]
#[repr(u8)]
pub enum ManifestContentType {
    /// Data.
    Data = 0,
    /// Deletes at position.
    PositionDeletes = 1,
    /// Delete by equality.
    EqualityDeletes = 2,
}
```

Reply:
> I think you have to distinguish between position deletes = 1 and equality deletes = 2. If we use the serde_repr crate we could directly serialize/deserialize it as follows:

Thanks for the reminder! I have a question:

In [spec of data file](https://iceberg.apache.org/spec/#manifests:~:text=int%20with%20meaning%3A%200%3A%20DATA%2C%201%3A%20POSITION%20DELETES%2C%202%3A%20EQUALITY%20DELETES), it distinguish position deletes and equality delete.
In [spec of manifest_list](https://iceberg.apache.org/spec/#manifests:~:text=int%20with%20meaning%3A%200%3A%20data%2C%201%3A%20deletes), it only distinguish data and delete.

So is that just a inconsistent in spec?

Reply:
Java's implementation also only has `Data` or `Delete` in manifest content type.

Reply:
> > I think you have to distinguish between position deletes = 1 and equality deletes = 2. If we use the serde_repr crate we could directly serialize/deserialize it as follows:
> 
> Thanks for the reminder! I have a question:
> 
> In [spec of data file](https://iceberg.apache.org/spec/#manifests:~:text=int%20with%20meaning%3A%200%3A%20DATA%2C%201%3A%20POSITION%20DELETES%2C%202%3A%20EQUALITY%20DELETES), it distinguish position deletes and equality delete. In [spec of manifest_list](https://iceberg.apache.org/spec/#manifests:~:text=int%20with%20meaning%3A%200%3A%20data%2C%201%3A%20deletes), it only distinguish data and delete.
> 
> So is that just a inconsistent in spec?

Sorry, my bad. I thought it was the same struct in `datafile`. But it looks like they are different. Maybe the information is not required in ManifestList. Forget my comment then.